src/Sarif/IResultLogWriter.cs

```diff
@@ -17,7 +33,15 @@ public interface IResultLogWriter
         /// <exception cref="ArgumentNullException">Thrown if <paramref name="info"/> is null.</exception>
         /// <param name="toolInfo">The tool information to write.</param>
         /// <param name="runInfo">The run information to write.</param>
-        void WriteToolAndRunInfo(ToolInfo toolInfo, RunInfo runInfo);
+        void WriteRunInfo(RunInfo runInfo);
+
+        /// <summary>
+        /// Initialize the results array associated with the current output log. SARIF producers that
+        /// are explicitly generating results (as opposed to other SARIF scenarios such as publishing
+        /// rules metadata) should proactively call this method in order to ensure that an explicit 
+        /// (but empty) results array exists in the log when no literal results were produced.
+        /// </summary>
+        void OpenResults();
```

Comment:
I'm a little concerned about this API. For one thing, SDK users might wonder why there's no `CloseResults` method. How about `CreateEmptyResultsArray`?